### PR TITLE
Adding in check for empty monoid

### PR DESCRIPTION
**Pull request by dalefrancis88**
This is a change to allow checking for `empty` monoids as requested by @jasminabasurita on https://github.com/evilsoft/crocks/issues/355

### CODE REVIEW

File: src/core/isEmpty.spec.js [hunk 1]
```diff
@@ -23,6 +28,7 @@ test('isEmpty predicate function', t => {
   t.equal(isEmpty(''), true, 'returns true with empty string')
   t.equal(isEmpty('string'), false, 'returns false with non-empty string')
 
+  t.equal(isEmpty(Fake.empty()), true, 'returns true with empty monoid')
```
Comment (karthikiyengar) [inline src/core/isEmpty.spec.js]: Might want to add `returns false for non-empty monoids` for consistency. I think that's what the CI is crying about as well?

File: src/core/isEmpty.js [hunk 2]
```diff
@@ -1,8 +1,14 @@
 /** @license ISC License (c) copyright 2016 original and current authors */
 /** @author Ian Hofmann-Hicks (evil) */
 const isObject = require('./isObject')
+const isMonoid = require('./isMonoid')
+const equals = require('./equals')
 
 function isEmpty(x) {
+  if(isMonoid(x)) {
+    return equals(x, x.empty())
```
Comment (evilsoft) [inline src/core/isEmpty.js]: to make this compatible with FL as well as crocks types, we should grab the empty off of the constructor doing `x.constructor.empty()`.

File: src/test/helpers.js [hunk 3]
```diff
@@ -27,7 +27,7 @@ function makeFake(algs, useFl) {
 
   const inst = xs.reduce((o, alg) => {
     const fn = useFl ? fl[alg] : alg
-    if(!isRepFunc(alg) && fn) {
+    if((!isRepFunc(alg) || alg === 'empty') && fn) {
```
Comment (evilsoft) [inline src/test/helpers.js]: Why is this needed?

File: src/core/isEmpty.js [hunk 4]
```diff
@@ -1,8 +1,14 @@
 /** @license ISC License (c) copyright 2016 original and current authors */
 /** @author Ian Hofmann-Hicks (evil) */
 const isObject = require('./isObject')
+const isMonoid = require('./isMonoid')
+const equals = require('./equals')
 
 function isEmpty(x) {
+  if(isMonoid(x)) {
+    return equals(x, x.constructor.empty())
```
Comment (evilsoft) [inline src/core/isEmpty.js]: Sorry, I totally forgot about fantasy land name prefixing,
One last change and then I think this will be GTG.

So this should be something like:
```js
const fl = require('../flNames')

return equals(x, (x.constructor[fl.empty] || x.empty)())
```

Comment (evilsoft) [reply src/core/isEmpty.js]: Actually to account for pre 1.x fl it should probably be:
```js
return equals(x, (x.constructor[fl.empty] || x.constructor.empty || x.empty)()
```

File: src/test/helpers.js (hunk 3 above)
Comment (dalefrancis88) [reply src/test/helpers.js]: This was needed because of the way that i was calling empty off of the instance. This code makes sure that the is an empty func on the instance and the constructor, however now that we've changed it this is very likely not needed. I just didn't think to remove it

**Comment by dalefrancis88** [comment]
Hey team, didn't realise there was an issue with the code coverage. Should be good to go now @evilsoft @HenriqueLimas
